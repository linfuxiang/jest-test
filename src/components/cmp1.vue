<template>
  <div >
    <h1>{{ msg }}</h1>
  </div>
</template>

<script>
export default {
  name: 'cmp1',
  data () {
    return {

    }
  },
  props: ['msg'],
  methods: {
  },
}
</script>

<style scope>

</style>
