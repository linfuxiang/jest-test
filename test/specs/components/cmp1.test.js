import { shallow } from 'vue-test-utils'
import Vue from 'vue'
import cmp1 from '@/components/cmp1'

// describe('cmp1.vue', () => {
// 	let cons, vm

// 	beforeAll(()=>{
// 		cons = Vue.extend(cmp1)
// 		vm = new cons().$mount()
// 	})
	
// 	it('test Data', () => {
// 		expect(vm.msg).toBe('Here is cmp1')
// 	})
// })

describe('just test cmp1.vue', ()=>{
	let cmp
	
	beforeAll(()=>{
		cmp = shallow(cmp1, {})
	})

	it('test Data', ()=>{
		expect(cmp.vm.msg).toBe('Here is cmp1')
	})
	it('snapshot', () => {
  		expect(cmp.element).toMatchSnapshot()
	})
})