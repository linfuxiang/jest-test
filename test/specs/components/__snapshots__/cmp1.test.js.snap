// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`just test cmp1.vue snapshot 1`] = `
<div>
  <h1>
    Here is cmp1
  </h1>
</div>
`;
